src/DeckViewer.jsx

```diff
@@ -1,76 +1,106 @@
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
 import { useNavigate } from 'react-router';
-import { shuffle, suitName } from './helpers';
+import { shuffle, suitName as suitNameFromSymbol } from './helpers';
 
-export default function DeckViewer() {
+export default function DeckViewer({ cards, jokers, title = 'Deck & Jokers' }) {
   const navigate = useNavigate();
-  const [deck, setDeck] = useState(() => generateMockDeck());
-  const [jokers] = useState(() => generateMockJokers());
-  const [query, setQuery] = useState('');
 
-  const filteredDeck = deck.filter(c => {
+  const isLiveMode = Array.isArray(cards) || Array.isArray(jokers);
+  const [mockDeck, setMockDeck] = useState(() => (isLiveMode ? [] : generateMockDeck()));
+  const [mockJokers] = useState(() => (isLiveMode ? [] : generateMockJokers()));
+
+  const deck = useMemo(() => {
+    const source = isLiveMode ? cards || [] : mockDeck;
+    return source.map(card =>
+      card && card.suitName ? card : { ...card, suitName: safeSuitName(card?.suit) }
+    );
+  }, [isLiveMode, cards, mockDeck]);
```

Comment:
I don’t really understand the point of this code. Is this a test? Why are we mocking instead of just using deck helpers to build cards?